src/libstore/derivations.hh

```diff
@@ -298,6 +298,10 @@ struct BasicDerivation
     std::string name;
 
     BasicDerivation() = default;
+    BasicDerivation(BasicDerivation &&) = default;
+    BasicDerivation(const BasicDerivation &) = default;
+    BasicDerivation& operator=(BasicDerivation &&) = default;
+    BasicDerivation& operator=(const BasicDerivation &) = default;
```

Comment:
If all the constructors are default, can't we just omit them?

Reply:
https://stackoverflow.com/questions/33957037/does-a-default-virtual-destructor-prevent-compiler-generated-move-operations

Reply:
Same with user-defined copy ctors for `ref`. A defined copy ctor deletes the move ctor/assignment operator. That's why it's better to not define anything by-hand if possible. In that case the compiler will do everything that the member types support.